Loop/Managers/DeviceDataManager.swift

```diff
@@ -1347,151 +1349,43 @@ extension Notification.Name {
     static let PumpEventsAdded = Notification.Name(rawValue:  "com.loopKit.notification.PumpEventsAdded")
 }
 
-// MARK: - Remote Notification Handling
-extension DeviceDataManager {
-    
-    func handleRemoteNotification(_ notification: [String: AnyObject]) {
-        Task {
-            let backgroundTask = await beginBackgroundTask(name: "Remote Data Upload")
-            await handleRemoteNotification(notification)
-            await endBackgroundTask(backgroundTask)
-        }
-    }
+// MARK: - ServicesManagerDosingDelegate
+
+extension DeviceDataManager: ServicesManagerDosingDelegate {
     
-    func handleRemoteNotification(_ notification: [String: AnyObject]) async {
+    func deliverBolus(amountInUnits: Double) async throws {
         
-        defer {
-            log.default("Remote Notification: Finished handling")
+        guard amountInUnits > 0 else {
```

Comment:
This validation could go into remote services manager, along with the error definitions. These are errors to report to the remote services, I think.  Won't block on this, though.

Reply:
This will put a dependency in ServicesManger on LoopSettings which sounds like you are ok with. I'll go ahead and make that change pre-merge. More contextual notes on the last changes are here https://loop.zulipchat.com/#narrow/stream/358458-Loop-Caregiver-App/topic/Merge.20Strategies/near/369827210

Reply:
I've made these changes here and don't have any planned changes. Let me know if anything is needed or merge if you are ready.